File: Src/java/engine/src/main/java/org/opencds/cqf/cql/engine/elm/execution/IdentifierRefEvaluator.java
package org.opencds.cqf.cql.engine.elm.execution;

import org.cqframework.cql.elm.execution.IdentifierRef;
import org.opencds.cqf.cql.engine.execution.Context;

public class IdentifierRefEvaluator extends IdentifierRef {

    @Override
    protected Object internalEvaluate(Context context) {

        String name = this.getName();

        if (name == null) {
            return null;
        }

        return context.resolveIdentifierRef(name);
    }
}

File: Src/java/engine/src/commonMain/kotlin/org/opencds/cqf/cql/engine/elm/executing/IdentifierRefEvaluator.kt
package org.opencds.cqf.cql.engine.elm.executing

import kotlin.jvm.JvmStatic
import org.opencds.cqf.cql.engine.execution.State
import org.opencds.cqf.cql.engine.runtime.Value

object IdentifierRefEvaluator {
    @JvmStatic
    fun internalEvaluate(name: kotlin.String?, state: State?): Value? {
        if (name == null) {
            return null
        }

        return state!!.resolveIdentifierRef(name)
    }
}
